src/PrestaShopBundle/Controller/Admin/ModuleController.php

```diff
@@ -325,23 +325,35 @@ public function importModuleAction(Request $request)
         $moduleManager = $this->get('prestashop.module.manager');
         try {
             $file_uploaded = $request->files->get('file_uploaded');
-            $violations = $this->get('validator')->validate($file_uploaded, new File(
-                [
+            $notNullViolations = $this->get('validator')->validateValue(
+                $file_uploaded,
+                new Assert\NotNull(array(
+                        'message' => $this->get('translator')->trans(
+                            'No file has been received. The server may have refused it because of its size.',
+                            array(),
+                            'Admin.Modules.Notification'))));
+
+            $fileViolations = $this->get('validator')->validateValue($file_uploaded, new Assert\File(
+                array(
+                    'disallowEmptyMessage' => 'An empty file is not allowed.',
```

Comment:
Why do you hardcode the message here ? You don't need it to "enable" the [violation](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Validator/Constraints/FileValidator.php#L209).

I'm pretty sure ``$request->files->get('file_uploaded')`` returns `null` despite the value needs to be an instance of ``UploadedFile``, this is why I'm ok with the second constraint :+1: